cdc server exits because of some RPC error

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

TiKV panic, and CDC processor exits

2. What did you expect to see?

3. What did you see instead?

> [2020/03/28 11:48:06.178 +08:00] [ERROR] [scheduler.go:71] ["error on running processor"] [captureid=4f9587b7-eeeb-4917-8200-1def460279e1] [changefeedid=f0eb97e5-d9a0-4c94-987a-f6c10f64b2b6] [processorid=a1744f94-4842-4374-87c1-efa856e5c511] [error="rpc error: code = Unavailable desc = timed out waiting for server handshake"] [errorVerbose="rpc error: code = Unavailable desc = timed out waiting for server handshake\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/juju_adaptor.go:15\ngithub.com/pingcap/ticdc/cdc/kv.(*CDCClient).dispatchRequest\n\tgithub.com/pingcap/ticdc/cdc/kv/client.go:393\ngithub.com/pingcap/ticdc/cdc/kv.(*CDCClient).EventFeed.func1\n\tgithub.com/pingcap/ticdc/cdc/kv/client.go:278\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57\nruntime.goexit\n\truntime/asm_amd64.s:1373"]

--- a/cdc/kv/client.go
+++ b/cdc/kv/client.go
@@ -394,10 +394,12 @@ MainLoop:
 			stream, ok := streams[rpcCtx.Addr]
 			// Establish the stream if it has not been connected yet.
 			if !ok {
-				// if get stream failed, maybe the store is down permanently, we should try to relocate the active store
 				stream, err = c.getStream(ctx, rpcCtx.Addr)
 				if err != nil {
+					// if get stream failed, maybe the store is down permanently, we should try to relocate the active store
 					log.Warn("get grpc stream client failed", zap.Error(err))
+					bo := tikv.NewBackoffer(ctx, tikvRequestMaxBackoff)
+					c.regionCache.OnSendFail(bo, rpcCtx, needReloadRegion(sri.failStoreIDs, rpcCtx), err)
 					continue MainLoop
 				}
 				streams[rpcCtx.Addr] = stream

--- a/cdc/kv/client.go
+++ b/cdc/kv/client.go
@@ -247,13 +247,19 @@ func (c *CDCClient) getConn(ctx context.Context, addr string) (*grpc.ClientConn,
 }
 
 func (c *CDCClient) getStream(ctx context.Context, addr string) (stream cdcpb.ChangeData_EventFeedClient, err error) {
-	conn, err := c.getConn(ctx, addr)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	client := cdcpb.NewChangeDataClient(conn)
-	stream, err = client.EventFeed(ctx)
-	log.Debug("created stream to store", zap.String("addr", addr))
+	err = retry.Run(500*time.Millisecond, 10, func() error {
+		conn, err := c.getConn(ctx, addr)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		client := cdcpb.NewChangeDataClient(conn)
+		stream, err = client.EventFeed(ctx)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		log.Debug("created stream to store", zap.String("addr", addr))
+		return nil
+	})
 	return
 }
 
@@ -388,9 +394,11 @@ MainLoop:
 			stream, ok := streams[rpcCtx.Addr]
 			// Establish the stream if it has not been connected yet.
 			if !ok {
+				// if get stream failed, maybe the store is down permanently, we should try to relocate the active store
 				stream, err = c.getStream(ctx, rpcCtx.Addr)
 				if err != nil {
-					return errors.Trace(err)
+					log.Warn("get grpc stream client failed", zap.Error(err))
+					continue MainLoop
 				}
 				streams[rpcCtx.Addr] = stream
 

